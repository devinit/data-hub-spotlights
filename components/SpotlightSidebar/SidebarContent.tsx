import React, { FunctionComponent } from 'react';

interface SidebarContentProps {
  height?: string;
}

const SidebarContent: FunctionComponent<SidebarContentProps> = ({ children, height }) => {
  return (
    <div className="spotlight__content" style={{ height }}>
      {children}
      <style jsx>{`
        height: 100%;
      `}</style>
    </div>
  );
};

export { SidebarContent };
