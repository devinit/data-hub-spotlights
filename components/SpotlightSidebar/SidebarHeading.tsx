import React, { FunctionComponent } from 'react';

interface SidebarHeadingProps {
  heading?: string;
  onClick?: () => void;
}

const SidebarHeading: FunctionComponent<SidebarHeadingProps> = ({ heading, onClick }) => {
  return (
    <h2 className="spotlight__heading">
      { heading }
      <button data-testid="spotlight-menu-trigger" type="button" className="button button--minor" onClick={ onClick }>
        <i role="presentation" aria-hidden="true" className="ico ico--20 ico-arrow-down-blank"/>
      </button>
    </h2>
  );
};

export { SidebarHeading };
