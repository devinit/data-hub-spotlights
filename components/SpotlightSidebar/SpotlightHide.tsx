import React, { FunctionComponent } from 'react';

interface SpotlightHideProps {
  height?: string;
}

const SpotlightHide: FunctionComponent<SpotlightHideProps> = ({ children, height }) => {
  return (
    <div className="spotlight__hide-ss" style={{ height }}>
      {children}
      <style jsx>{`
        height: 100%;
      `}</style>
    </div>
  );
};

export { SpotlightHide };
