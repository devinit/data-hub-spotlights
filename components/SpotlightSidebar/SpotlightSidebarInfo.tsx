import React, { FunctionComponent } from 'react';

interface SpotlightSidebarInfoProps {
  heading?: string;
  description?: string;
}

const SpotlightSidebarInfo: FunctionComponent<SpotlightSidebarInfoProps> = (props) => {
  return (
    <div className="spotlight-description">
      <h3 className="spotlight__subheading">{props.heading}</h3>
      <p className="spotlight__excerpt">{props.description}</p>
    </div>
  );
};

export { SpotlightSidebarInfo };
