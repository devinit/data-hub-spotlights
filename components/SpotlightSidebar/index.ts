export * from './SpotlightSidebar';
export * from './SidebarContent';
export * from './SidebarHeading';
