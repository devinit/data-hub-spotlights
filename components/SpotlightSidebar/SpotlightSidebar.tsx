import React, { Children, FunctionComponent, isValidElement } from 'react';
import { SidebarContent } from './SidebarContent';
import { SidebarHeading } from './SidebarHeading';

const SpotlightSidebar: FunctionComponent = ({ children }) => {

  return (
    <div className="spotlight__aside spotlight__aside--padded">
      {
        Children.map(children, child =>
          isValidElement(child) && (child.type === SidebarContent || child.type === SidebarHeading) ? child : null)
      }
      <style jsx>{ `
        min-height: 500px;
      ` }</style>
    </div>
  );
};

export { SpotlightSidebar };
