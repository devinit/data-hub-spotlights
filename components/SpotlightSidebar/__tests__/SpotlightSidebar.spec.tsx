/**
 * @jest-environment jsdom
 */
import '@testing-library/jest-dom/extend-expect';
import React from 'react';
import * as TestRenderer from 'react-test-renderer';
// import { render } from '@testing-library/react';
import { SpotlightSidebar } from '../SpotlightSidebar';

describe('SpotlightSidebar', () => {
  test('renders correctly', () => {
    const renderer = TestRenderer.create(<SpotlightSidebar />).toJSON();

    expect(renderer).toMatchSnapshot();
  });
});
