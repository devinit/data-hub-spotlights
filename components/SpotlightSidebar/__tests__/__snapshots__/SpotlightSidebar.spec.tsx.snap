// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpotlightSidebar renders correctly 1`] = `
<div
  className="spotlight__aside spotlight__aside--padded"
/>
`;
